lib/foods.js

```diff
@@ -4,37 +4,92 @@
 const $ = require("jQuery")
 
+$('#newFoodForm').hide();
+
 function getFoods() {
   //counter = 1;
   $.ajax({
     type: "GET",
     url: herokuUrl() + "api/v1/foods",
     success: function(posts) {
-      posts.forEach(function(foodType){
+      posts.reverse().forEach(function(foodType){
         event.preventDefault;
         let button = `<td><button type='button' class='deleteButton' id='${foodType.id}'>Delete</button></td>`
           let toInsert = `<tr id=${foodType.id}><td>${foodType.name}</td><td>${foodType.calories}</td>${button}</tr>`;
-          $(".foodsTable").append(toInsert);
+        $(".foodsTable").append(toInsert);
       });
       deleteFoodListener();
+      createFoodButtonListener();
+      createFoodFormListener();
     }
   })
 }
 
 
 function deleteFoodListener() {
-$('.deleteButton').on('click', function() {
-  console.log('delete button pressed');
-  //debugger;
-
-$.ajax({
-  type: 'DELETE',
-  url: herokuUrl() + `api/v1/foods/${event.target.id}`,
-  success: $(`#${event.target.id}`).hide()
+  $('.deleteButton').on('click', function() {
+
+    $.ajax({
+      type: 'DELETE',
+      url: herokuUrl() + `api/v1/foods/${event.target.id}`,
+      success: $(`#${event.target.id}`).hide()
+    });
+  });
+};
+
+function createFoodButtonListener() {
+  $('#createButton').on('click', function() {
+    $('#newFoodForm').show();
   });
-});
 };
 
+function createFoodFormListener() {
+  $('#createFoodForm').on('submit', function(event) {
+    event.preventDefault();
+
+    let food = $(event.currentTarget.name).val();
+    let calories = $(event.currentTarget.calories).val();
+
+    errorCheck(food, calories);
+
+    let data = $.post(herokuUrl() + `api/v1/foods?name=${food}&calories=${calories}"`)
+      .then(function(data) {
+        let button = `<td><button type='button' class='deleteButton' id='${data.id}'>Delete</button></td>`
+          let toInsert = `<tr id=${data.id}><td>${food}</td><td>${calories}</td>${button}</tr>`;
+        $(".foodsTable").prepend(toInsert);
+        $('#createFoodForm')[0].reset();
+      });
+  });
+};
+
+function errorCheck(food, calories) {
+  //let words = [food, calories];
+    //let toSay = {
+      //food: "food name",
+      //calories: "calorie amount"
+    //}
+
+  //words.forEach(function(input, index) {
+    //let word = ["food", "calories"];
+    //if (!`${input}`) {
+      //$(`${word[index]}-error`).remove();
+      //throw new Error ($(`#${word[index]}Field`).append(`<p id=``${word[index]}-error``>Please enter a ${toSay[word[index]]} </p>`));
+    //};
+  //});
+//};
+
+    if (!food) {
+      $('#name-error').remove();
+      throw new Error ($('#nameField').append("<p id='name-error'>Please enter a food name</p>"));
+    }
+
+    if (!calories || calories < 1) {
+      $('#calorie-error').remove();
+      throw new Error($('#calorieField').append("<p id='calorie-error'>Please enter a calorie amount</p>"));
+    };
+};
+
+
 getFoods();
```

Comment:
So...Many...commented lines... haha great job Bret, these look sound. There are a handful of refactoring opportunities for us moving forward but overall this looks great 🥇